### ISSUE TITLE
Cannot use IssueFields stuct as valid 'fields' object when performing a transition w/ payload

### ISSUE THREAD

**Issue by acidbee** [andygrunwald/go-jira#105]
The fields `Type`/`issuetype` and `Summary` do not have `omitempty` in their stuct tags. This is certainly fine for a create, but when performing a transition, the presence of these fields without data in them confuses the JIRA API and error response of 400 is returned.

Example of a marshaled request using IssueFields for `fields`:
```
rest/api/2/issue/SAMPLE-123/transitions
{
	"update": {
		"comment": [{
			"add": {
				"body": "bug has been fixed."
			}
		}]
	},
	"fields": {
		"issuetype": {},
		"summary": ""
                "resolution": {
			"self": "",
			"id": "",
			"description": "",
			"name": "Done"
		},
		"customfield_14610": {
			"value": "Complete"
		}
	},
	"transition": {
		"id": "21",
		"name": "Close"
	}
}
```

Response:
```
{
    "errorMessages": [],
    "errors": {
        "summary": "Field 'summary' cannot be set. It is not on the appropriate screen, or unknown.",
        "issuetype": "Could not find issuetype by id or name.",
    }
}
```

**Comment by andygrunwald** [comment]
Can you post a minimal code example to reproduce this? Thank you.

**Comment by acidbee** [comment]
```go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"

	"github.com/andygrunwald/go-jira"
	"github.com/trivago/tgo/tcontainer"
)

type TransitionPayload struct {
	Update        *tcontainer.MarshalMap `json:"update,omitempty"`
	Fields        *jira.IssueFields      `json:"fields,omitempty"`
	Transition    *Transition            `json:"transition"`
}

type Transition struct {
	Id    string        `json:"id,omitempty"`
	Name  string        `json:"name,omitempty"`
}

func main() {
	resolution := jira.Resolution{}
	resolution.Name = "Fixed"
	resolution.Description = "Some description"

	issueFields := jira.IssueFields{}
	issueFields.Resolution = &resolution

	transition := Transition{}
	transition.Id = "21"
	transition.Name = "Close"

	payload := TransitionPayload{}
	payload.Transition = &transition
	payload.Fields = &issueFields

    client, _ := jira.NewClient(nil, "https://your.jira.url")
    client.Authentication.SetBasicAuth("john.doe", "redacted")

    sampleBody, _ := json.Marshal(payload)
    fmt.Println(string(sampleBody))

    response, err := client.Issue.DoTransitionWithPayload("SAMPLE-123", payload)
	if err != nil {
		unexpectedResponse(response, err)
	}

	fmt.Println("Tansition Successful")
}

func unexpectedResponse(response *jira.Response, err error) {
	if response != nil && response.Body != nil {
		// With an unexpected response there MAY be something still in the body buffer, try to get it.
		buf := new(bytes.Buffer)
		defer response.Body.Close()
		buf.ReadFrom(response.Body)
		body := buf.String()
		panic(fmt.Errorf("%s :: %s", err.Error(), body))
	} else {
		panic(err)
	}
}
```

**Comment by acidbee** [comment]
I'd also request adding support for the update structure in Issue for PUTs, in the short term at least as a MarshalMap:

`Update  *tcontainer.MarshalMap  'json:"update,omitempty" structs:"update,omitempty"'`

**Comment by andygrunwald** [comment]
Hey @acidbee, sorry for the late response.
I am right now limited in time due to some private things, i will come back to you later, ok?

@rbriski Can you help here?

**Comment by rbriski** [comment]
OK,  I do see what @acidbee is talking about.  The docs are here:
https://developer.atlassian.com/cloud/jira/platform/rest/#api-api-2-issue-issueIdOrKey-transitions-post

The specific line that describes this behavior is:
`If a field is not configured to appear on the transition screen, it will not be returned in the transition metadata. A field validation error will occur if such field is submitted in issue transition request.`

Let me work on writing a failing test for this.

@acidbee Regarding `MarshalMap`, is this related to the `omitempty` issue?  If not, can you create another issue and explain why you want this functionality?  Or PRs are always welcome :)

**Comment by rbriski** [comment]
Wait, this was already fixed is #110.  I'm going to close this issue.

**Comment by acidbee** [comment]
@rbriski Sorry about that. I probably should've included a test that covers that specific issue. But my PR did patch the two fields in IssueFields.
